var/spack/repos/builtin/packages/conduit/package.py

```diff
@@ -107,10 +107,10 @@ class Conduit(Package):
     #
     # Use HDF5 1.8, for wider output compatibly
     # variants reflect we are not using hdf5's mpi or fortran features.
-    depends_on("hdf5@1.8.19:1.8.999~cxx~mpi~fortran", when="+hdf5+hdf5_compat+shared")
-    depends_on("hdf5@1.8.19:1.8.999~shared~cxx~mpi~fortran", when="+hdf5+hdf5_compat~shared")
-    depends_on("hdf5~cxx~mpi~fortran", when="+hdf5~hdf5_compat+shared")
-    depends_on("hdf5~shared~cxx~mpi~fortran", when="+hdf5~hdf5_compat~shared")
+    depends_on("hdf5@1.8.19:1.8.999~cxx", when="+hdf5+hdf5_compat+shared")
+    depends_on("hdf5@1.8.19:1.8.999~shared~cxx", when="+hdf5+hdf5_compat~shared")
+    depends_on("hdf5~cxx", when="+hdf5~hdf5_compat+shared")
+    depends_on("hdf5~shared~cxx", when="+hdf5~hdf5_compat~shared")
```

Comment:
@cyrush, I was not able to build `mfem+petsc+conduit` without the above modifications -- `petsc` needs `hdf5+mpi+fortran` but `conduit` asks for `hdf5~mpi~fortran`.

Are you OK with the above modifications, or would you prefer another solution?

Reply:
@v-dobrev thanks the context, I am fine with those changes since they are needed for this case. 

Conduit does not need a version of hdf5 that has MPI support, however being too restrictive here causes spack nightmares - I fully appreciate that. Since we use our own spack fork for all of our testing, this change wont cause any problems for us. 

In the future, we (Conduit folks) may just need to give guidance about the spack solution de jour (environments, etc) for dialing in the hdf5 options.

Reply:
OK, thanks @cyrush!